### DISCUSSION TITLE
Disable all linting in pylance

### DISCUSSION THREAD

**Discussion by stevenlis** [microsoft/pylance-release#4249] [Q&A] [answered]
Is there any way to disable all linting in pylance. Right now i just do it in `python.analysis.diagnosticSeverityOverrides`. 
```
    "python.analysis.diagnosticSeverityOverrides": {
        "reportMissingImports": "none",
        "reportUndefinedVariable": "none"
    }
    
```
Do I must spell out all the rules here:
https://github.com/microsoft/pyright/blob/main/docs/configuration.md

Is there a easy way to do so?

**Comment by rchiodo** [comment, marked as answer]
You could set your language server to none? 
```json
"python.languageServer": "None"
```
Or do you still want completions but not have any errors show up?

To remove more errors you'd either have to do what you're already doing, or perhaps switch the `typeCheckingMode` to off if you haven't already:

```json
"python.analysis.typeCheckingMode": "off"
```

**Reply by stevenlis** [reply]
Hi, @rchiodo. Thanks for the quick reply. After read the [doc](https://github.com/microsoft/pyright/blob/main/docs/configuration.md) more carefully, I realized that if I set `typeCheckingMode` to off, there are only a few settings still left on. so I did:
```json
    "python.analysis.typeCheckingMode": "off",
    "python.analysis.diagnosticSeverityOverrides": {
        "strictListInference": "none",
        "strictDictionaryInference": "none",
        "strictSetInference": "none",
        "analyzeUnannotatedFunctions": "none",
        "strictParameterNoneValue": "none",
        "enableTypeIgnoreComments": "none",
        "reportMissingModuleSource": "none",
        "reportMissingImports": "none",
        "reportUndefinedVariable": "none",
    }
```

I'm actually switching to Ruff. Sometimes the jupyter extension has bugs that will affect pylance, and jupyter extension only get released once a month now, which is quite annoying that you have to wait for a month for a bug to be fixed.

**Reply by erictraut** [reply]
Ruff is a linter whereas Pylance is a language server and type checker. They serve different purposes and are largely complementary. If you're looking for a linter, Ruff is a good choice. If you're looking for completion suggestions, auto-import, signature help, refactoring tools, static type checking, etc., Pylance provides those capabilities.

To disable all diagnostic rules in Pylance, you can use:
```
    "python.analysis.typeCheckingMode": "off",
    "python.analysis.diagnosticSeverityOverrides": {
        "reportMissingModuleSource": "none",
        "reportMissingImports": "none",
        "reportUndefinedVariable": "none",
    }
```
You included a few extra items that are not diagnostic rules and therefore have no effect in the `diagnosticSeverityOverrides` setting.

**Reply by stevenlis** [reply]
Thanks @erictraut. I didn't realize that i overdid it. Indeed, I do like the auto-completion provided by pylance so only want to disable its linting functions. In terms of `auto-import`... I turned it off a long time ago and couldn't remember what the problem is, and it's even off by default now. For type checker... pylance has to use the stubs in each package I used, and those stub files might not be complete... so eventually I'm so frustrated and had to turn it off as well... When the auto-completion works, it works very well... but too often it does not... I have no idea why... There are many things that make no sense to me. Here's a very simple example:

my cursor is already in this function and I don't know why the autocompletion have to show me anything but the params of the function.
<img width="478" alt="CS-zH9Pa37t@2x" src="https://user-images.githubusercontent.com/33796896/232629250-818f5c93-185b-405f-8ecf-16f886ab731a.png">


and then, it seems like, it knows the param takes a boolen...
<img width="497" alt="CS-M3MJWJYy@2x" src="https://user-images.githubusercontent.com/33796896/232628875-21fce6bc-0210-4b0e-a4a7-530a5d3c4cd8.png">

and somehow.... this is what it shows you:


<img width="615" alt="CS-9nqhC3Pk@2x" src="https://user-images.githubusercontent.com/33796896/232629000-391b2f77-5b2b-4a3a-8382-8da9c8233d76.png">

Why... Error...and fig???

**Reply by rchiodo** [reply]
> my cursor is already in this function and I don't know why the autocompletion have to show me anything but the params of the function.

Pylance is showing all the things you could possibly pass as that argument that match the letters you've typed. We could probably do a better job limiting them. Might be too slow though. Limiting the completions to those things that match the type of the argument you're passing. That wouldn't work so good if there were overloads or if we had to parse a lot of symbols to determine their return value.

**Reply by stevenlis** [reply]
Thanks for the explanation @rchiodo. Hope there would be a solution for this one day.

**Comment by xshapira** [comment]
@stevenlis 
Adding this to `settings.json` disables only the linting part of pylance:

```json
"python.analysis.ignore": ["*"],
```

[stackoverflow.com/a/75071893](https://stackoverflow.com/a/75071893)

**Reply by amzarmuzani** [reply]
> "python.analysis.ignore": ["*"],

Note that the above setting will also deactivate Pylance's type-checking capabilities.
